tools/isopack.js

```diff
@@ -1100,18 +1100,64 @@ _.extend(Isopack.prototype, {
         !f.match(/^examples\/unfinished/);
     });
 
-    var gitSha = files.runGitInCheckout('rev-parse', 'HEAD');
-
+    // Regexes matching paths to transpile using babel
+    var transpileRegexes = [
+      /^tools\/[^\/]+\.js$/, // General tools files
+      // We don't support running self-test from an install anymore
+    ];
+
+    // Split pathsToCopy into two arrays - one of files that should be copied
+    // directly, and one of files that should be transpiled with Babel
+    var pathsToTranspile = [];
+    var pathsToCopyStraight = [];
+    pathsToCopy.forEach((path) => {
+      var shouldTranspile =
+        _.some(transpileRegexes, (regex) => path.match(regex));
+
+      if (shouldTranspile) {
+        pathsToTranspile.push(path);
+      } else {
+        pathsToCopyStraight.push(path);
+      }
+    });
 
+    // Set up builder to write to the correct directory
     var toolPath = 'mt-' + archinfo.host();
     builder = builder.enter(toolPath);
+
+    // Transpile the files we selected
+    var babel = require("babel");
```

Comment:
That would mean that we would have to avoid using es6 in any of the files in the tool that are part of Isobuild, or come  up with some brand new clever ways of avoiding the circular dependency. (Other than JSAnalyze, all of the other isopackets are used by parts of the tool other than the core build logic.)